docs/devguide.md

```diff
@@ -297,6 +298,36 @@ If you choose third party resources storage, the helm chart will not launch an
 etcd server, but will instead instruct the API server to store all resources in
 the Kubernetes cluster as third party resources.
 
+## Dependency Management
+We use [dep](https://golang.github.io/dep) manage our dependencies. We commit the resulting
+vendor directory to ensure repeatable builds and isolation from upstream source disruptions.
+Because vendor is committed, you do not need to interact with dep unless you are
+changing dependencies.
+
+* Gopkg.toml - the dep manifest, this is intended to be hand-edited and contains a set of
+constraints and other rules for dep to apply when selecting appropriate versions of dependencies.
+* Gopkg.lock - the dep lockfile (generated). For the most part, don't edit this file because dep
+completely overwrites that file during `dep ensure`. If you run into trouble with this file, ask.
```

Comment:
ask who?

Reply:
Can we summarize the rule?

Reply:
> ask who?

In the PR or our slack channel? Should I put my phone number in there? 😉 

> Can we summarize the rule?

I wish I could! Let me try.... 🤔 

"You may hand edit Gopkg.lock when to instruct dep to select a specific revision on a branch that is not the HEAD of the branch while continuing to use a branch constraint in Gopkg.toml, which will make it easier to upgrade and track the branch in the future (as opposed to pinning the revision in a constraint)."

But that is such an advanced case that I'm not sure it's worth mentioning. Maybe just stick to "don't edit this file".

Reply:
I would suggest to keep it "don't edit lock file".
The advanced case you described is about developing a branch which we don't have any restrictions for. But once the PR against master is open, we want to keep it reproducible and not allow manual editing.